### PR TITLE
Refactor {{each-in}} tests

### PR DESCRIPTION
**Pull request by cibernox**
Replaces #15579

### CODE REVIEW

File: packages/ember-glimmer/tests/integration/syntax/each-in-test.js [hunk 1]
```diff
@@ -57,222 +62,231 @@ applyMixins(BasicEachInTest,
...
+
+    this.assertText('Smartphones: 8203JavaScript Frameworks: Infinity');
 
     this.assertStableRerender();
 
-    this.runTask(() => {
-      set(this.context, 'categories.Smartphones.reports.unitsSold', 100);
-      set(this.context, 'categories.Tweets', { reports: { unitsSold: 443115 } });
-    });
+    if (this.allowsSetProp) { // Not al backing data structures allow kvo tracking. Maps and Iterables don't
```
Comment (cibernox) [inline packages/ember-glimmer/tests/integration/syntax/each-in-test.js]: @chancancode This is the idea I came up with to allow to run the same tests for every backing data structure but skip one specific part on data structures that do not allow KVO tracking: Maps and Iterables.

I considered (and I actually started with that approach) to define a `setProp` implementation that essentially replaced the Map/Iterable with a modified one, in an immutable fashino, but I had the impression that it was hiding the fact from the reader that the feature wasn't actually supported for some objects.

**Comment by cibernox** [comment]
@chancancode Ready

**Comment by cibernox** [comment]
This is finally green. Dat prettier refactor...

**Comment by rwjblue** [comment]
Sorry @cibernox :(

**Pull request merged**